lib/native/error.ts

```diff
@@ -21,8 +21,12 @@ import { isNumber } from 'util';
  * a human-readable string.
  */
 export class CrtError extends Error {
+    public readonly error_code?: number;
     constructor(err: any) {
```

Comment:
probably also want to save the AWS_ERROR_ENUM_NAME because honestly if they're handling specific errors that's the easiest one to check against. ex:
`if (error.name == "AWS_ERROR_SOCKET_TIMEOUT")`

But ugh.. an enum would be better.
And a definitive list of possible values would be EVEN BETTER but that would involve some codegen... or an O(N) crawl over possible error values at boot to build up some dynamic list.

ramble ramble

Reply:
I can add that. Can I do it as a follow up before this PR kills someone?

Reply:
yes. absolutely